lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -981,6 +981,345 @@ int parse_ncellmeas(const char *at_response, struct lte_lc_cells_info *cells)
+		err = 1;
+		LOG_DBG("NCELLMEAS status %d", status);
+		goto clean_exit;
+	}
+
+	/* Go through the cells. */
+	for (i = 0; curr_index < (param_count - (AT_NCELLMEAS_GCI_CELL_PARAMS_COUNT + 1)) &&
+			i < params->gci_count; i++) {
+		struct lte_lc_cell parsed_cell;
+
+		/* <cell_id>  */
+		curr_index++;
+		err = string_param_to_int(&resp_list, curr_index, &tmp, 16);
+		if (err) {
+			LOG_ERR("Could not parse cell_id, index %d, i %d error: %d",
+				curr_index, i, err);
+			goto clean_exit;
+		}
+
+		if (tmp > LTE_LC_CELL_EUTRAN_ID_MAX) {
+			tmp = LTE_LC_CELL_EUTRAN_ID_INVALID;
+		}
+		parsed_cell.id = tmp;
+
+		/* <plmn> */
+		len = sizeof(tmp_str);
+
+		curr_index++;
+		err = at_params_string_get(&resp_list, curr_index,
+				tmp_str, &len);
+		if (err) {
+			LOG_ERR("Could not parse plmn, error: %d", err);
+			goto clean_exit;
+		}
+		tmp_str[len] = '\0';
+
+		/* Read MNC and store as integer. The MNC starts as the fourth character
+		 * in the string, following three characters long MCC.
+		 */
+		err = string_to_int(&tmp_str[3], 10, &parsed_cell.mnc);
+		if (err) {
+			LOG_ERR("string_to_int, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* Null-terminated MCC, read and store it. */
+		tmp_str[3] = '\0';
+
+		err = string_to_int(tmp_str, 10, &parsed_cell.mcc);
+		if (err) {
+			LOG_ERR("string_to_int, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <tac> */
+		curr_index++;
+		err = string_param_to_int(&resp_list, curr_index, &tmp, 16);
+		if (err) {
+			LOG_ERR("Could not parse tac in i %d, error: %d", i, err);
+			goto clean_exit;
+		}
+		parsed_cell.tac = tmp;
+
+		/* <ta> */
+		curr_index++;
+		err = at_params_int_get(&resp_list, curr_index,
+					&tmp);
+		if (err) {
+			LOG_ERR("Could not parse ta, error: %d", err);
+			goto clean_exit;
+		}
+		parsed_cell.timing_advance = tmp;
+
+		/* <ta_meas_time> */
+		curr_index++;
+		err = at_params_int64_get(&resp_list, curr_index,
+					  &parsed_cell.timing_advance_meas_time);
+		if (err) {
+			LOG_ERR("Could not parse ta_meas_time, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <earfcn> */
+		curr_index++;
+		err = at_params_int_get(&resp_list, curr_index,
+					&parsed_cell.earfcn);
+		if (err) {
+			LOG_ERR("Could not parse earfcn, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <phys_cell_id> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.phys_cell_id);
+		if (err) {
+			LOG_ERR("Could not parse phys_cell_id, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <rsrp> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.rsrp);
+		if (err) {
+			LOG_ERR("Could not parse rsrp, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <rsrq> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.rsrq);
+		if (err) {
+			LOG_ERR("Could not parse rsrq, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <meas_time> */
```

Comment:
Is meas_time only interesting for determining the time difference of measurements or is there some other use case for this? We could calculate the time difference of measurements instead or in addition to meas_time. I wonder why we get time difference for legacy PCI measurements but not for GCI.

Reply:
I would keep this as plain meas_time in here. I think that timeDiff should be determined in a code that does an actual use case, in this case in multicell_lib or in nrf cloud lib. We don't necessarily know all actual use cases.
<meas_time> is only known for the serving cell, that is not know for other GCI cells.